From 0ab86a2301d850a6add7b09784d8745f3e6344e3 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Fri, 24 Apr 2026 21:33:10 +0000
Subject: [PATCH 180/220] WIP: drm/rockchip: dw_hdmi: Enable ycbcr444/420 and
 10-bit for RK3228/RK3328

NOTE: enable supported formats on rk3228/rk3328
NOTE: split in two?

Signed-off-by: Jonas Karlman <jonas@kwiboo.se>
---
 drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c b/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c
index 40d4af351a04..3ad1db9f1ac0 100644
--- a/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c
+++ b/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c
@@ -474,6 +474,10 @@ static const struct dw_hdmi_plat_data rk3228_hdmi_drv_data = {
 	.phy_ops = &rk3228_hdmi_phy_ops,
 	.phy_name = "inno_dw_hdmi_phy2",
 	.phy_force_vendor = true,
+	.supported_formats = BIT(DRM_OUTPUT_COLOR_FORMAT_RGB444) |
+			     BIT(DRM_OUTPUT_COLOR_FORMAT_YCBCR444) |
+			     BIT(DRM_OUTPUT_COLOR_FORMAT_YCBCR420),
+	.max_bpc = 10,
 };
 
 static struct rockchip_hdmi_chip_data rk3288_chip_data = {
@@ -510,7 +514,11 @@ static const struct dw_hdmi_plat_data rk3328_hdmi_drv_data = {
 	.phy_ops = &rk3328_hdmi_phy_ops,
 	.phy_name = "inno_dw_hdmi_phy2",
 	.phy_force_vendor = true,
+	.supported_formats = BIT(DRM_OUTPUT_COLOR_FORMAT_RGB444) |
+			     BIT(DRM_OUTPUT_COLOR_FORMAT_YCBCR444) |
+			     BIT(DRM_OUTPUT_COLOR_FORMAT_YCBCR420),
 	.use_drm_infoframe = true,
+	.max_bpc = 10,
 };
 
 static struct rockchip_hdmi_chip_data rk3368_chip_data = {
-- 
2.43.0

